CMakeLists.txt

```diff
@@ -51,6 +51,14 @@ endif()
 set(NVIM_VERSION_MAJOR 0)
 set(NVIM_VERSION_MINOR 1)
 set(NVIM_VERSION_PATCH 1)
+# Package maintainers should define this according to their package system
+# conventions, if the source code or runtime files were modified, e.g.:
+#   cmake -DNVIM_VERSION_PRERELEASE=-deb3
+# or define it as empty to prevent "-dev", if no modifications were made:
+#   cmake -DNVIM_VERSION_PRERELEASE=
+if(NOT DEFINED NVIM_VERSION_PRERELEASE)
+  set(NVIM_VERSION_PRERELEASE "-dev")
+endif()
```

Comment:
> If a source snapshot is not a release, it should have "-dev", otherwise it shouldn't.

AFAIK that is how this PR functions. In which case would that not be the case (except in a git tree)? 

> Semi-related: currently we number versions like: 0.1.0 -> 0.1.0-dev -> 0.1.1 -> 0.1.1-dev -> ... This doesn't follow SemVer:

Ok, what if we did `0.1.0 -> 0.1.1-dev -> 0.1.1 -> 0.1.2-dev -> ...` ?

> libuv more or less at random.. looks like they do what I described: tagging a release, follow-up commit

I tend to look at git, which AFAICT does not do that. We've been burned before by following node.js habits (e.g.: CLA). How about other projects (e.g., Rust, Atom--I did not check these myself)?

> git describe complicates the situation somewhat. We don't really follow SemVer anymore since we introduced that. Hm.

Section 9 of semver.org says arbitrary pre-release data may be appended after a hyphen...